vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1547,12 +1547,24 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
     }
 
     void finish_load() final {
+        auto& device_ctx = g_vpr_ctx.mutable_device();
         process_rr_node_indices();
 
         rr_graph_builder_->init_fan_in();
+        /* Create a temp copy to convert from vtr::vector to std::vector
+         * This is required because the ``alloc_and_load_rr_indexed_data()`` function supports only std::vector data
+         * type for ``rr_segments``
+         * Note that this is a dirty fix (to avoid massive code changes)
+         * TODO: The ``alloc_and_load_rr_indexed_data()`` function should embrace ``vtr::vector`` for ``rr_segments``
+         */
+        std::vector<t_segment_inf> temp_rr_segs;
+        temp_rr_segs.reserve(segment_inf_.size());
+        for (auto& rr_seg : device_ctx.rr_segments) {
```

Comment:
The ``device_ctx.rr_segments`` is already cached in this data structure as ``segment_inf_``. You do not need to call the ``device_ctx`` here.

Reply:
How we will replace device.rr_segments then here ?